### PR TITLE
Simulation: More Robust Stats

### PR DESCRIPTION
**Pull request by itanasi**
Save the information in each step object so we can run more robust stats post-processing on it

### CODE REVIEW

**Comment by itanasi** [comment]
Still feels a bit hacky so any suggestions appreciated

File: core/src/com/unciv/logic/simulation/Simulation.kt [hunk 1]
```diff
@@ -30,6 +30,8 @@ class Simulation(
     var steps = ArrayList<SimulationStep>()
     var numWins = mutableMapOf<String, MutableInt>()
     private var summaryStats = HashMap<String, HashMap<Int, HashMap<String, MutableInt>>>() // [civ][turn][stat]=value
+    private val keySum = "sum"
```
Comment (yairm210) [inline core/src/com/unciv/logic/simulation/Simulation.kt:33]: At this point, why not make this an enum?

Comment (itanasi) [reply core/src/com/unciv/logic/simulation/Simulation.kt]: Yeah, good idea.

**Comment by EmperorPinguin** [comment]
How will this deal with CS stats? Currently the population counts doesn't work you add city states to the simulation, it'd be better if these are ignored.

**Comment by itanasi** [comment]
> How will this deal with CS stats? Currently the population counts doesn't work you add city states to the simulation, it'd be better if these are ignored.

Right now it records the CS civ info, but since the final Prints are only whoever wins (which by definition Major Civs), they're never printed.

Do you want the stats for CS printed too?

**Comment by EmperorPinguin** [comment]
> Do you want the stats for CS printed too?

No, I don't need those stats, it just crashed when I added in city states when also enabling the stat collection. I'll see if I can reproduce it tomorrow.

**Comment by itanasi** [comment]
Anyways, this is fine to pull unless someone has more changes. I tried running with CS and it didn't crash in this new version.

**Pull request merged**